sormas-backend/src/main/java/de/symeda/sormas/backend/immunization/ImmunizationFacadeEjb.java

```diff
@@ -274,6 +282,17 @@ private Immunization fillOrBuildEntity(@NotNull ImmunizationDto source, Immuniza
 		return target;
 	}
 
+	@Override
+	public void updateImmunizationStatuses() {
```

Comment:
Ah, I guess my development specifications were misleading in this regard - I didn't mean to write a native query, just wanted to point out that the entities should not be loaded, changed, and then persisted again, but instead there should be a direct update. I would still go with CriteriaQuery here unless there's a good reason not to; see e.g. CaseService.updateArchived as an example.

Additionally, please make sure to update the change date of the affected immunizations as well.